overlord/configstate/config/helpers.go

```diff
@@ -145,7 +145,8 @@ func GetSnapConfig(st *state.State, snapName string) (*json.RawMessage, error) {
 func SetSnapConfig(st *state.State, snapName string, snapcfg *json.RawMessage) error {
 	var config map[string]*json.RawMessage
 	err := st.Get("config", &config)
-	isNil := snapcfg == nil || len(*snapcfg) == 0
+	// empty nil snapcfg should be an empty message, but deal with "null" as well.
+	isNil := snapcfg == nil || len(*snapcfg) == 0 || bytes.Compare(*snapcfg, []byte("null")) == 0
```

Comment:
If I remove the "bytes.Compare()" here and run the (unit and snapshot-basic spread) tests nothing changes:
```
index 737421270d..563c5c49ae 100644
--- a/overlord/configstate/config/helpers.go
+++ b/overlord/configstate/config/helpers.go
@@ -146,7 +146,7 @@ func SetSnapConfig(st *state.State, snapName string, snapcfg *json.RawMessage) e
        var config map[string]*json.RawMessage
        err := st.Get("config", &config)
        // empty nil snapcfg should be an empty message, but deal with "null" as well.
-       isNil := snapcfg == nil || len(*snapcfg) == 0 || bytes.Compare(*snapcfg, []byte("null")) == 0
+       isNil := snapcfg == nil || len(*snapcfg) == 0
        if err == state.ErrNoState {
                if isNil {
                        // bail out early
```
so either:
a) I did something wrong
b) this is not needed
c) our tests need to improve

Reply:
Great catch! The check in the test was not sufficient, added now. Thanks!